tests/test_all.py

```diff
@@ -763,3 +768,203 @@ def test_container_suseconnect_adds_repos(container_per_test: ContainerData):
     container_per_test.connection.check_output("zypper -n ref")
     repos = get_repos_from_connection(container_per_test.connection)
     assert len(repos) > 3
+
+
+_USERNAME_UID_GID_MAP: Dict[str, Tuple[Optional[int], Optional[int]]] = {
+    "nobody": (65534, 65534),
+    "root": (0, 0),
+    "wwwrun": (None, 485),
+    "pesign": (None, 486),
+    "nginx": (None, 486),
+    "registry": (None, 486),
+    "app": (1654, 1654),
+    "mysql": (60, 60),
+    "dirsrv": (None, 486),
+    "postgres": (None, 486),
+    "pcp": (496, 484),
+    "ldap": (498, 498),
+    "systemd-coredump": (497, 100),
+    "postfix": (51, 51),
+    "keadhcp": (None, 486),
+    "user": (1000, 1000),
+}
+
+# special cases for TW & SLE 16
+if OS_VERSION in ("tumbleweed", "16.0"):
+    # these users don't use the non-default GID on TW
+    for username in (
+        "nginx",
+        "dirsrv",
+        "postgres",
+        "registry",
+        "keadhcp",
+    ):
+        del _USERNAME_UID_GID_MAP[username]
+
+    # completely different UID & GID on TW
+    _USERNAME_UID_GID_MAP["pcp"] = (496, 498)
+    _USERNAME_UID_GID_MAP["wwwrun"] = (None, 498)
+    _USERNAME_UID_GID_MAP["pesign"] = (499, 499)
+    _USERNAME_UID_GID_MAP["systemd-coredump"] = (497, 1000)
+
+
+@pytest.mark.parametrize("container", ALL_CONTAINERS, indirect=True)
+def test_uids_stable(container: ContainerData) -> None:
+    """Check that every user in :file:`/etc/passwd` has a stable uid & gid as
+    defined in ``_USERNAME_UID_GID_MAP``.
+
+    """
+    # collect users who owns subdirectories in directories /var, /etc, /opt, home
+    user_list = (
+        container.connection.check_output(
+            "find /var /etc /opt /home -printf '%u \n' | sort -u"
+        )
+        .strip()
+        .split("\n")
+    )
+    passwd: str = container.connection.file("/etc/passwd").content_string
+    assert container.connection.user("root").exists, "root user does not exist"
+
+    for userline in passwd.splitlines():
+        tmp = userline.split(":")
+        name, uid, gid = tmp[0], int(tmp[2]), int(tmp[3])
+        if name not in user_list:
+            continue
+
+        expected_uid, expected_gid = _USERNAME_UID_GID_MAP.get(
+            name, (499, 499)
+        )
+        expected_uid = 499 if expected_uid is None else expected_uid
+        expected_gid = 499 if expected_gid is None else expected_gid
+
+        # special cases
+        if (container.container.get_base().baseurl or "").split(":")[
+            0
+        ].endswith("kiosk/xorg") and name == "user":
+            expected_gid = 100
+
+        if (
```

Comment:
Can we move this to a fixture? There's `_USERNAME_UID_GID_MAP` for generic users, maybe `_USERNAME_UID_GID_IMAGE_MAP` could map image names vs users for special cases.

It would remove these IFs here and be in one single place. To update in the future, it would be easier.